at_client/lib/src/service/notification_service_impl.dart

```diff
@@ -408,4 +412,37 @@ class NotificationServiceImpl
   getStreamListenersCount() {
     return _streamListeners.length;
   }
+
+  @override
+  Future<AtNotification> fetch(String notificationId) async {
+    var notifyFetchVerbBuilder = NotifyFetchVerbBuilder()
+      ..notificationId = notificationId;
+    var atNotificationStr = await _atClient
+        .getRemoteSecondary()
+        ?.executeVerb(notifyFetchVerbBuilder);
+    if (atNotificationStr == null) {
+      throw AtClientException.message('Failed to fetch the notification id',
+          intent: Intent.remoteVerbExecution,
+          exceptionScenario: ExceptionScenario.remoteVerbExecutionFailed);
+    }
+    AtResponse atResponse = DefaultResponseParser().parse(atNotificationStr);
+    var atNotificationMap = jsonDecode(atResponse.response);
+    if (atNotificationMap['notificationStatus'] ==
+        at_persistence_secondary_server.NotificationStatus.expired.toString()) {
+      return AtNotification.empty()
```

Comment:
Why the special handling for 'expired' status? i.e. why not just pass the full notification back regardless of status?

Reply:
Oh I think I know the answer to my question - it's because the server only returns the id and the status in this case.